rotkehlchen/api/rest.py

```diff
@@ -1041,7 +1041,17 @@ def add_history_event(self, event: HistoryBaseEntry) -> Response:
         return api_response(result_dict, status_code=HTTPStatus.OK)
 
     def edit_history_event(self, event: HistoryBaseEntry) -> Response:
+        # The extra_data field is something that we don't expose to the frontend in the case of
+        # evm transactions because is a field that only we edit. In this case we need to be sure
+        # at the moment of editing an event that this information is kept in the database and
+        # doesn't get overwritten. The identifier shoul not be None here because is sent by the
+        # frontend to edit the events
+        if event.identifier is None:
```

Comment:
this cannot happen as this check happens at the serializer level.
https://github.com/rotki/rotki/blob/f701f372e95bf1a7925bb87983cad382efe63486/rotkehlchen/api/v1/schemas.py#L615